sdk/identity/identity/src/credentials/managedIdentityCredential/cloudShellMsi.ts

```diff
@@ -53,14 +53,21 @@ function prepareRequestOptions(
 
 /**
  * Defines how to determine whether the Azure Cloud Shell MSI is available, and also how to retrieve a token from the Azure Cloud Shell MSI.
+ * Since Azure Managed Identities aren't available in the Azure Cloud Shell, we log a warning for users that try to access cloud shell using user assigned identity.
```

Comment:
I like this! Please add a changelog entry?

Reply:
Please also add a test! 🙂 in the file for ManagedIdentityCredential tests located in the `test/internal/node` folder.